src/Utilities/Compiler/CodeMetrics/CodeMetricsAnalysisContext.cs

```diff
@@ -0,0 +1,33 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+#if HAS_IOPERATION
+
+using System;
+using System.Collections.Concurrent;
+using System.Threading;
+
+namespace Microsoft.CodeAnalysis.CodeMetrics
+{
+    public sealed class CodeMetricsAnalysisContext
```

Comment:
I have tried different ways of passing down the user option to the named type metric calculator and this one was IMO the best solution in term of reading and evolution. Please feel free to suggest another approach if you don't like this one.